### PR TITLE
Feature/502

### PR DESCRIPTION
**Pull request by M1ndForce**
#502

### CODE REVIEW

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 1]
```diff
@@ -3,138 +3,146 @@ import Button from '@material-ui/core/Button';
...
 import Grid from '@material-ui/core/Grid';
+import { useHistory, useLocation } from 'react-router';
 import PriceFilter from './price-filter';
 import HotItemFilter from './hot-item-filter';
 
 import { useStyles } from './product-list-filter.styles';
 import {
   getFiltredProducts,
-  setColorsFilter,
+  // setColorsFilter,
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:13]: Please remove this commented line

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 2]
```diff
@@ -3,138 +3,146 @@ import Button from '@material-ui/core/Button';
...
 } from '../../../redux/products/products.actions';
 
 import {
   MODEL_TEXT,
   PATTERN_TEXT,
   CATERGORY_TEXT,
-  CLEAR_FILTER_BUTTON_TEXT,
-  COLORS_TEXT
+  CLEAR_FILTER_BUTTON_TEXT
+  // COLORS_TEXT
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:24]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 3]
```diff
@@ -3,138 +3,146 @@ import Button from '@material-ui/core/Button';
...
       language: Language.language,
-      currency: Currency.currency,
-      filterStatus: Products.filterStatus
+      filterData: Products.filterData
     })
   );
   const {
     categoryFilter,
-    colorsFilter,
+    // colorsFilter,
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:43]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 4]
```diff
@@ -3,138 +3,146 @@ import Button from '@material-ui/core/Button';
...
     patternsFilter,
     modelsFilter
   } = filters;
+  //
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:47]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 5]
```diff
@@ -3,138 +3,146 @@ import Button from '@material-ui/core/Button';
...
-      filterHandler: (e) => handleFilterChange(e, setModelsFilter, modelsFilter)
-    },
-    colors: {
-      filterName: COLORS_TEXT[language].value,
-      productFilter: colorsFilter,
-      list: colorsNames,
-      clearFilter: () => handleFilterClear(setColorsFilter),
-      filterHandler: (e) => handleFilterChange(e, setColorsFilter, colorsFilter)
+      list: filterData.models.map((model) => model.name[language].value),
+      categories: filterData.models, // add this for working
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:113]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js [hunk 6]
```diff
@@ -3,138 +3,146 @@ import Button from '@material-ui/core/Button';
...
+      filterAction: setModelsFilter,
+      labels: 'modelsFilter',
+      clearFilter: () => handleFilterClear(setModelsFilter, 'modelsFilter'),
+      filterHandler: (e) =>
+        handleFilterChange(e, 'modelsFilter', filterData.models)
     },
+    // colors: {
```
Comment (moran711) [inline src/pages/product-list-page/product-list-filter/product-list-filter.js:120]: same

File: src/pages/product-list-page/product-sort/product-sort.js [hunk 7]
```diff
@@ -12,38 +13,61 @@ import {
...
     language: Language.language
   }));
   const styles = useStyles();
   const dispatch = useDispatch();
+  const history = useHistory();
+  const { search } = useLocation();
+  const searchParams = new URLSearchParams(search);
+  const { sort, page, defaultPage } = URL_QUERIES_NAME;
+  const query = searchParams.get(sort);
+  // const sortAction = [setSortByPrice, setSortByRate, setSortByPrice];
```
Comment (moran711) [inline src/pages/product-list-page/product-sort/product-sort.js:29]: same

File: src/pages/product-list-page/product-sort/product-sort.js [hunk 8]
```diff
@@ -12,38 +13,61 @@ import {
...
+  useEffect(() => {
+    // SORT_BY_SELECT_OPTIONS.forEach(({ optionValue }, index) => {
```
Comment (moran711) [inline src/pages/product-list-page/product-sort/product-sort.js:31]: same

File: src/redux/products/products.reducer.js [hunk 9]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
-    categoryFilter: null,
+    categoryFilter: [],
     priceFilter: [0, 99999999],
     searchFilter: '',
     modelsFilter: [],
     isHotItemFilter: false
   },
-  filterData: [],
+  filterData: {
+    categories: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:48]: I think it would be better to set empty array

File: src/redux/products/products.reducer.js [hunk 10]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
+          {
+            value: ''
+          },
+          {
+            value: ''
+          }
+        ]
+      }
+    ],
+    models: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:61]: same

File: src/redux/products/products.reducer.js [hunk 11]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
+          {
+            value: ''
+          },
+          {
+            value: ''
+          }
+        ]
+      }
+    ],
+    patterns: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:74]: same

File: src/redux/products/products.reducer.js [hunk 12]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
+          {
+            value: ''
+          },
+          {
+            value: ''
+          }
+        ]
+      }
+    ],
+    closures: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:98]: same

File: src/redux/products/products.reducer.js [hunk 13]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
+          {
+            value: ''
+          },
+          {
+            value: ''
+          }
+        ]
+      }
+    ],
+    mainMaterial: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:122]: same

File: src/redux/products/products.reducer.js [hunk 14]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
+          {
+            value: ''
+          },
+          {
+            value: ''
+          }
+        ]
+      }
+    ],
+    mainMaterialColor: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:135]: same

File: src/redux/products/products.reducer.js [hunk 15]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
+          {
+            value: ''
+          },
+          {
+            value: ''
+          }
+        ]
+      }
+    ],
+    innerMaterialColor: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:148]: same

File: src/redux/products/products.reducer.js [hunk 16]
```diff
@@ -38,13 +38,162 @@ export const initialState = {
...
+          {
+            value: ''
+          },
+          {
+            value: ''
+          }
+        ]
+      }
+    ],
+    bottomMaterial: [
```
Comment (moran711) [inline src/redux/products/products.reducer.js:172]: same

File: src/redux/products/products.sagas.js [hunk 17]
```diff
@@ -91,7 +91,7 @@ export function* handleProductLoading({ payload }) {
 }
 
 export default function* productsSaga() {
-  yield takeEvery(GET_ALL_FILTERS, handleGetAllProducts);
+  yield takeEvery(GET_ALL_FILTERS, handleGetAllProductsFilters); // here new saga - handleGetAllProductsFilters
```
Comment (moran711) [inline src/redux/products/products.sagas.js:94]: same

File: src/redux/products/products.types.js [hunk 18]
```diff
@@ -2,7 +2,7 @@ export const SET_PRODUCT = 'SET_PRODUCT';
 export const GET_PRODUCT = 'GET_PRODUCT';
 export const SET_PRODUCT_LOADING = 'SET_PRODUCT_LOADING';
 export const SET_ALL_PRODUCTS = 'SET_ALL_PRODUCTS';
-export const GET_ALL_FILTERS = 'GET_ALL_PRODUCTS';
+export const GET_ALL_FILTERS = 'GET_ALL_PRODUCTS'; // before was GET_ALL_PRODUCTS
```
Comment (moran711) [inline src/redux/products/products.types.js:5]: Please change this string to `GET_ALL_FILTERS `

File: src/redux/products/products.sagas.js [hunk 19]
```diff
@@ -54,15 +52,17 @@ export function* handleFilteredProductsLoad({ payload: { forSearchBar } }) {
...
     }
   } catch (e) {
+    console.log(e);
     yield call(handleProductsErrors, e);
   }
 }
 
-export function* handleGetAllProducts() {
+export function* handleGetAllProductsFilters() {
+  // refactor saga
```
Comment (moran711) [inline src/redux/products/products.sagas.js:61]: same

File: src/pages/product-list-page/product-list-page.js [hunk 20]
```diff
@@ -60,43 +62,61 @@ const ProductListPage = ({ category, model, width }) => {
...
+    isHotItemFilter
+  } = filters;
 
   useEffect(() => {
     dispatch(getAllFilters());
   }, [dispatch]);
 
+  const fetchFilteredData = useCallback(
+    (dispatched) => {
+      console.log('callback');
```
Comment (moran711) [inline src/pages/product-list-page/product-list-page.js:82]: Please remove this

File: src/pages/product-list-page/product-list-page.js [hunk 21]
```diff
@@ -60,43 +62,61 @@ const ProductListPage = ({ category, model, width }) => {
...
-          ...filterData.map((product) => product.basePrice[currency].value)
-        )
-      ])
-    );
-  }, [category, filterData, model, currency, dispatch]);
-
-  useEffect(() => {
-    dispatch(getFiltredProducts({}));
+    fetchFilteredData(dispatch);
+    console.log('useEffect');
```
Comment (moran711) [inline src/pages/product-list-page/product-list-page.js:103]: sa,e

File: src/pages/product-list-page/product-list-page.js [hunk 22]
```diff
@@ -60,43 +62,61 @@ const ProductListPage = ({ category, model, width }) => {
...
-        Math.max(
-          ...filterData.map((product) => product.basePrice[currency].value)
-        )
-      ])
-    );
-  }, [category, filterData, model, currency, dispatch]);
-
-  useEffect(() => {
-    dispatch(getFiltredProducts({}));
+    fetchFilteredData(dispatch);
```
Comment (moran711) [inline src/pages/product-list-page/product-list-page.js:102]: Why do you need to set dispatch in this function

File: src/pages/product-list-page/product-list-page.js [hunk 23]
```diff
@@ -60,43 +62,61 @@ const ProductListPage = ({ category, model, width }) => {
...
+  } = filters;
 
   useEffect(() => {
     dispatch(getAllFilters());
   }, [dispatch]);
 
+  const fetchFilteredData = useCallback(
+    (dispatched) => {
+      console.log('callback');
+      dispatched(setCurrentPage(searchParams.get('page')));
```
Comment (moran711) [inline src/pages/product-list-page/product-list-page.js:83]: Please use `dispatch` func

File: src/pages/product-list-page/product-list-page.js [hunk 24]
```diff
@@ -60,43 +62,61 @@ const ProductListPage = ({ category, model, width }) => {
...
+      dispatched(getFiltredProducts({}));
```
Comment (moran711) [inline src/pages/product-list-page/product-list-page.js:84]: same

File: src/redux/products/products.sagas.js [hunk 25]
```diff
@@ -54,15 +52,17 @@ export function* handleFilteredProductsLoad({ payload: { forSearchBar } }) {
       yield put(setProductsLoading(false));
     }
   } catch (e) {
+    console.log(e);
```
Comment (moran711) [inline src/redux/products/products.sagas.js:55]: same

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 1 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js:13]: fixed

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 2 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js:24]: fixed

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 3 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js:43]: fixed

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 4 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js:47]: fixed

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 5 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js:113]: fixed

File: src/pages/product-list-page/product-list-page.js (hunk 21 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-page.js:103]: fixed

File: src/pages/product-list-page/product-list-page.js (hunk 22 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-page.js:102]: fixed

File: src/pages/product-list-page/product-list-page.js (hunk 23 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-page.js:83]: fixed

File: src/pages/product-list-page/product-list-page.js (hunk 24 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-page.js:84]: fixed

File: src/redux/products/products.sagas.js (hunk 25 above)
Comment (M1ndForce) [reply src/redux/products/products.sagas.js:55]: fixed

File: src/pages/product-list-page/product-list-filter/product-list-filter.js (hunk 6 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-list-filter/product-list-filter.js]: fixed

File: src/pages/product-list-page/product-sort/product-sort.js (hunk 7 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-sort/product-sort.js]: fixed

File: src/pages/product-list-page/product-sort/product-sort.js (hunk 8 above)
Comment (M1ndForce) [reply src/pages/product-list-page/product-sort/product-sort.js]: fixed

File: src/redux/products/products.reducer.js (hunk 9 above)
Comment (M1ndForce) [reply src/redux/products/products.reducer.js]: fixed

File: src/redux/products/products.sagas.js (hunk 19 above)
Comment (M1ndForce) [reply src/redux/products/products.sagas.js]: fixed

File: src/redux/products/products.sagas.js (hunk 17 above)
Comment (M1ndForce) [reply src/redux/products/products.sagas.js]: fixed

File: src/redux/products/products.types.js (hunk 18 above)
Comment (M1ndForce) [reply src/redux/products/products.types.js]: fixed

**Pull request closed without merging**